### PR TITLE
feat: YostarEN Reclamation2

### PR DESCRIPTION
**Pull request by zayn7lie**

### CODE REVIEW

**Comment by HX3N** [comment]
You can refer to [this comment](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/9992#issuecomment-2267951824) for ocr text in other mode

**Review by Constrat — APPROVED** [review]
Just did a run with it. Everything worked flawlessly.
@zayn7lie Quick question. Are you using 1920x1080?
In case you're not, somehow everything still worked on my 1920x1080 environment (which I suggest as some texts are stupidly small sometimes)

**Comment by HX3N** [comment]
Wait a second, it seems like `Reclamation2ExReadArchiveSelectDayConfirm` is missing

**Comment by Constrat** [comment]
I was about to check the Ex one now. I need to understand how the item craft are interpreted in the core. Directly ocred? Like with IS starting operator?

**Comment by HX3N** [comment]
> I was about to check the Ex one now. I need to understand how the item craft are interpreted in the core. Directly ocred? Like with IS starting operator?

It recognizes the equipment list using OCR, selects the identified equipment, and proceeds with the crafting. (If the OCR recognition is inaccurate, it seems like it doesn't work)

**Comment by Constrat** [comment]
Yeah I guess exactly like the IS one. You need the client language to fill it. No `CharsNameOcrReplace` is done

**Comment by zayn7lie** [comment]
Wait a sec. I am still debugging with that.

Since mode 1 requires enough level and days in that, I need some time to reach the requirement for translation.

**Comment by Constrat** [comment]
Yeah I noticed that something is needed to use the Ex correctly :P

**Comment by zayn7lie** [comment]
> Just did a run with it. Everything worked flawlessly. @zayn7lie Quick question. Are you using 1920x1080? In case you're not, somehow everything still worked on my 1920x1080 environment (which I suggest as some texts are stupidly small sometimes)

Nope. I am using 1280x720. I think it is because of OCR could also handle with some scaling, rotating or so.

At the same time, WordOCR seems to do nothing much with resolution lol.

**Comment by Constrat** [comment]
The issue is when \n are present. If you check this you will understand why I've been only using 1920 for the past year and a half: 
- https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/5235

**Comment by zayn7lie** [comment]
> The issue is when `\n` are present. If you check this you will understand why I've been only using 1920 for the past year and a half:
> 
>     * [Resolution Inconsistency: AutoSquad Copilot Operator's name Detection in YostarEN #5235](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/5235)
> 
> 
> Luckily this time the wording seems identical in both 720 and 1080 but that's usually the opposite.

I think there might be one way to handle with that, just separate the terms into words?

**Comment by Constrat** [comment]
Most alters use the same words and usually when there's a \n the and the text is small the OCR does not consider the two words apart and mixes them leading to a unusable result.

**Comment by zayn7lie** [comment]
> Most alters use the same words and usually when there's a \n the and the text is small the OCR does not consider the two words apart and mixes them leading to a unusable result.

NP, I will also handle with that later

**Comment by Constrat** [comment]
Remember that too many ocrreplace will crash MAA as it already happened in the JP client with the Trader in IS.

**Comment by Constrat** [comment]
image
Nvm I see why. Probably still haven't unlocked something aye

**Comment by HX3N** [comment]
image

**Comment by Constrat** [comment]
"Reclamation2ExReadArchiveEnter": {
    "algorithm": "OcrDetect",
    "action": "ClickSelf",
    "text": ["读取"],
    "roi": [1030, 548, 170, 124],
    "sub": ["Reclamation2ExReadArchiveSelectDay", "Reclamation2ExReadArchiveSelectDayConfirm"]
},

This is 

"Reclamation2ExReadArchiveEnter": {
    "text": ["Load Save"],
},

**Comment by zayn7lie** [comment]
> image
> 
> should unlock the shop. It will probably open at base level 3

If it's a good idea to crop the middle of them?

**Comment by Constrat** [comment]
I don't think the shop is needed in any of the actions for manufacturing. Increasing the ROI should not be a problem.

File: resource/global/YoStarEN/resource/tasks.json [hunk 1]
```diff
@@ -2559,12 +2559,29 @@
...
         "text": ["Craft"]
     },
+    "Reclamation2ExManufactureConfirmAmount": {
+        "algorithm": "OcrDetect",
+        "Doc": "如果刚好用完的话可能会卡住，这个之后再处理了",
+        "text": ["99"],
+        "roi": [720, 442, 175, 166],
+        "sub": ["Reclamation2ExManufactureConfirm"],
+        "next": ["Reclamation2ExManufacture"]
+    },
```
Comment (Constrat) [inline resource/global/YoStarEN/resource/tasks.json:2569]: This can be removed

**Comment by zayn7lie** [comment]
> I don't think the shop is needed in any of the actions for manufacturing. Increasing the ROI should not be a problem.

Fine. The potential problem of different resolution might only happen on "Are you sure you want /n to leave?". For wordOCR and cropping it only scan "Are you sure you" so it will not cause a big problem lol. But anyway let me also handle with that.

**Comment by zayn7lie** [comment]
NP. Feel free to have a check~

**Comment by HX3N** [comment]
> I don't think the shop is needed in any of the actions for manufacturing. Increasing the ROI should not be a problem.

It seems to be because the task needs to be started when the game is already quite progressed. It wouldn't matter if the reclamation is initiated as soon as it is recognized, but wouldn't there be an issue when trying to load the saved survival days later?

**Comment by Constrat** [comment]
No file changes are with ROI so let's leave it as is, without trying to center the ROI to make it work for both ways.

@HX3N I noticed you are missing two tasks on the korean?
    "Reclamation2ExBegin": {
        "text": ["Resume"]
    },
    
    and
    
    
  "Reclamation2ExReadArchiveConfirm": {
      "text": ["Confirm"]
  },

**Comment by zayn7lie** [comment]
> > I don't think the shop is needed in any of the actions for manufacturing. Increasing the ROI should not be a problem.
> 
> It seems to be because the task needs to be started when the game is already quite progressed. It wouldn't matter if the reclamation is initiated as soon as it is recognized, but wouldn't there be an issue when trying to load the saved survival days later? (The Ex task involves repeating the process of crafting > loading saved survival days)

Do you mean the feature of "Archive Days"? I have done with "Archive Days" (or "Save" in game). Check [#10143](https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/10143/commits/c770d1579540dbd642f9109e3efd66d4343d6ef2)

**Comment by zayn7lie** [comment]
> No file changes are with ROI so let's leave it as is, without trying to center the ROI to make it work for both ways.
> 
> @HX3N I noticed you are missing two tasks on the korean?
> 
> ```
>     "Reclamation2ExBegin": {
>         "text": ["Resume"]
>     },
> ```
> 
> and
> 
> ```
>   "Reclamation2ExReadArchiveConfirm": {
>       "text": ["Confirm"]
>   },
> ```

Day 4 is enough for "Save" feature. You could check with those ocr quickly~

**Comment by Constrat** [comment]
Anyway we can merge. As most features are done.
We can always fix them later. To also clearup the discussion.

@zayn7lie Thank you very much :)

**Comment by Constrat** [comment]
@HX3N I would like to push a nightly. Do you want to add the missing tasks and run the nightly yourself? Or do I just run it now?

**Comment by HX3N** [comment]
> @HX3N I would like to push a nightly. Do you want to add the missing tasks and run the nightly yourself? Or do I just run it now?

do update. I have encountered a new problem, so it might take a while.

**Comment by dontrunmk** [comment]
is RA2 currently available on the stable version?

**Comment by HX3N** [comment]
> is RA2 currently available on the stable version?

Not possible in the stable version yet. It is currently only available in the Nightly version.

**Pull request merged**